src/github.com/stellar/horizon/db2/history/ledger_cache.go

```diff
@@ -0,0 +1,47 @@
+package history
+
+import (
+	"github.com/stellar/go/support/errors"
+)
+
+// Queue adds `seq` to the load queue for the cache.
+func (lc *LedgerCache) Queue(seq int32) {
+	lc.lock.Lock()
+
+	if lc.queued == nil {
+		lc.queued = map[int32]struct{}{}
+	}
+
+	lc.queued[seq] = struct{}{}
+	lc.lock.Unlock()
```

Comment:
That's certainly a valid behavior, but I decided against it since the ledger cache is only used on a per-request basis for now.  Every use will be "queue up a bunch of sequences, hit load once, then throw the results away".  Perhaps `LedgerBatcher` would be a better name?

I was nervous about instituting a global cache that has better reuse possibilities;  Well nervous about including it in this PR anyway.  I'm open to renaming the struct though... what makes most sense to you?